gate-deck-plugins/src/main/kotlin/com/netflix/spinnaker/gate/plugins/DeckPluginService.kt

```diff
@@ -66,6 +67,23 @@ class DeckPluginService(
     }
     registry.counter(assetHitsId).increment()
 
-    return localAsset.readText()
+    return PluginAsset.from(localAsset)
+  }
+
+  data class PluginAsset(val contentType: String, val content: String) {
+
+    companion object {
+      fun from(file: File): PluginAsset {
+        return PluginAsset(
+            contentType = when {
+              file.toString().endsWith(".js") -> { "application/javascript" }
+              file.toString().endsWith(".css") -> { "text/css" }
+              file.toString().endsWith(".html") -> { "text/html" }
+              else -> "text/plain"
```

Comment:
We should log something when this default occurs, I think.

Reply:
Good call, made it log at warn level - seems like something we should pay attention to.